<launch>
    <arg name="robot"                   default="robot1"/>
    <arg name="x"                       default="0.0"/>
    <arg name="y"                       default="0.0"/>
    <arg name="marker_size"             default="14"/>
    <arg name="max_new_marker_error"    default="0.08"/>
    <arg name="max_track_error"         default="0.2"/>
    <arg name="cam_image_topic"         default="/camera/rgb/image_raw"/>
    <arg name="cam_info_topic"          default="/camera/rgb/camera_info"/> 
    <arg name="output_frame"            default="/camera_rgb_optical_frame"/> 
    <node pkg = "ar_track_alvar" type = "individualMarkersNoKinect" respawn = "false" output = "screen" name = "ar_track_alvar">
        <param name = "marker_size"             type = "double"         value = "$(arg marker_size)"/>
        <param name = "max_new_marker_error"    type = "double"         value = "$(arg max_new_marker_error)"/>
        <param name = "max_track_error"         type = "double"         value = "$(arg max_track_error)"/>
        <param name = "output_frame"            type = "string"         value = "$(arg output_frame)"/>

        <remap from = "camera_image" to = "$(arg cam_image_topic)"/>
        <remap from = "camera_info"  to = "$(arg cam_info_topic)"/>
    </node>

    <node pkg = "aruco_listener"  type = "aruco_listener" name = "aruco_listener_$(arg robot)" output="screen">
        <param name = "robot_name"    type = "string"         value = "$(arg robot)"/>
        <param name = "startX"        type = "double"         value = "$(arg x)"/>
        <param name = "startY"        type = "double"         value = "$(arg y)"/>

        <!--pid value, for test, don't change it-->
        <param name = "kp_q"            type = "double"         value = "1.58"/>
        <param name = "ki_q"            type = "double"         value = "0.0"/>
        <param name = "kd_q"            type = "double"         value = "0.047"/>
        <param name = "kp_v"            type = "double"         value = "0.9"/>
        <param name = "ki_v"            type = "double"         value = "0.01"/>
        <param name = "kd_v"            type = "double"         value = "0.2"/>
        <param name = "kp_yaw"            type = "double"         value = "0.012"/>
        <param name = "ki_yaw"            type = "double"         value = "0.002"/>
        <param name = "kd_yaw"            type = "double"         value = "0.14"/>

    </node>
</launch>
